src/components/MovieForm/MovieForm.tsx

```diff
@@ -1,36 +1,67 @@
-import { ChangeEvent, useState } from 'react';
+import React, { ChangeEvent, useEffect, useState } from 'react';
 
 import './MovieForm.css';
 
 import { Movie } from '../../models/movie';
 import Button from '../Button/Button';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { addMovie, updateMovie } from '../../services/api.service';
+import { createSearchParams, useNavigate, useSearchParams } from 'react-router-dom';
 
 export type MovieFormProps = {
-    movie?: Movie | null;
-    onSubmit: (editedMovie?: Movie) => void;
-    onReset: () => void;
+    movie: Movie | undefined;
 };
 
-const MovieForm = ({ movie, onReset, onSubmit }: MovieFormProps) => {
+const MovieForm = ({ movie }: MovieFormProps) => {
+    const navigate = useNavigate();
+    const {
+        register,
+        handleSubmit,
+        formState: { errors },
+        reset,
+    } = useForm<Movie>();
+
+    const [searchParams] = useSearchParams();
     const [editedMovie, updateEditedMovie] = useState<Movie>(movie ?? ({} as Movie));
 
-    const handleSubmitForm = (event: any) => {
-        console.log(event);
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-member-access
-        event?.preventDefault();
-        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument,@typescript-eslint/no-unsafe-member-access
-        onSubmit(Object.fromEntries(new FormData(event?.target)) as unknown as Movie);
+    useEffect(() => {
+        updateEditedMovie(movie ?? ({} as Movie));
+        if (movie) {
+            reset(movie);
+        }
+    }, [movie]);
+
+    const onSubmit: SubmitHandler<Movie> = async (data: Movie) => {
+        const addedMovie = movie ? await updateMovie(data) : await addMovie(data);
```

Comment:
Just recommendation here. Difference between updated and new movie is ID , maybe is more stronger to use movie.id for condition instead entire object.